evm/tests/erc721.rs

```diff
@@ -0,0 +1,308 @@
+use plonky2_evm::generation::{GenerationInputs, TrieInputs};
+use plonky2_evm::proof::{BlockHashes, BlockMetadata, TrieRoots};
+use plonky2_evm::prover::prove;
+use plonky2_evm::verifier::verify_proof;
+use plonky2_evm::Node;
+
+type F = GoldilocksField;
+const D: usize = 2;
+type C = KeccakGoldilocksConfig;
+
+/// Test a simple ERC721 token transfer.
+/// Used the following Solidity code:
+/// ```solidity
+/// pragma solidity ^0.8.20;
+///
+/// import "@openzeppelin/contracts@5.0.1/token/ERC721/ERC721.sol";
+/// import "@openzeppelin/contracts@5.0.1/access/Ownable.sol";
+///
+/// contract TestToken is ERC721, Ownable {
+///     constructor(address initialOwner)
+///         ERC721("TestToken", "TEST")
+///         Ownable(initialOwner)
+///     {}
+///
+///     function safeMint(address to, uint256 tokenId) public onlyOwner {
+///         _safeMint(to, tokenId);
+///     }
+/// }
+/// ```
+///
+/// The transaction calls the `safeTransferFrom` function to transfer token `1337` from address
+/// `0x5B38Da6a701c568545dCfcB03FcB875f56beddC4` to address `0xAb8483F64d9C6d1EcF9b849Ae677dD3315835cb2`.
+#[test]
+fn test_erc721() -> anyhow::Result<()> {
+    init_logger();
+
+    let all_stark = AllStark::<F, D>::default();
+    let config = StarkConfig::standard_fast_config();
+
+    let beneficiary = hex!("deadbeefdeadbeefdeadbeefdeadbeefdeadbeef");
+    let owner = hex!("5B38Da6a701c568545dCfcB03FcB875f56beddC4");
+    let contract = hex!("f2B1114C644cBb3fF63Bf1dD284c8Cd716e95BE9");
+
+    let owner_state_key = keccak(owner);
+    let contract_state_key = keccak(contract);
+
+    let owner_nibbles = Nibbles::from_bytes_be(owner_state_key.as_bytes()).unwrap();
+    let contract_nibbles = Nibbles::from_bytes_be(contract_state_key.as_bytes()).unwrap();
+
+    let mut state_trie_before = HashedPartialTrie::from(Node::Empty);
+    state_trie_before.insert(owner_nibbles, rlp::encode(&owner_account()).to_vec());
+    state_trie_before.insert(contract_nibbles, rlp::encode(&contract_account()).to_vec());
+
+    let storage_tries = vec![(contract_state_key, contract_storage())];
+
+    let tries_before = TrieInputs {
+        state_trie: state_trie_before,
+        transactions_trie: HashedPartialTrie::from(Node::Empty),
+        receipts_trie: HashedPartialTrie::from(Node::Empty),
+        storage_tries,
+    };
+
+    let txn = signed_tx();
+
+    let gas_used = 58_418.into();
+    let block_metadata = BlockMetadata {
+        block_beneficiary: Address::from(beneficiary),
+        block_timestamp: 0x03e8.into(),
+        block_number: 1.into(),
+        block_difficulty: 0x020000.into(),
+        block_random: H256::from_uint(&0x020000.into()),
+        block_gaslimit: 0xff112233u32.into(),
+        block_chain_id: 1.into(),
+        block_base_fee: 0xa.into(),
+        block_gas_used: gas_used,
+        // Technically wrong Bloom filter, but it's not checked by the prover anyway.
+        block_bloom: [0.into(); 8],
```

Comment:
nit: what's the rationale of not filling it up with what's expected past this txn execution, for consistency at least?